src/main/java/at/ac/fhcampuswien/fhmdb/HomeController.java

```diff
@@ -13,104 +13,160 @@
-        genreComboBox.setValue(Genre.ALL);
-        if (genreComboBox.getValue() != null) {
-            genreComboBox.getSelectionModel().clearSelection();
+
+        sortBtn.setText(SORT_DEFAULT_TEXT_DESC);
+
+        genreComboBox.setValue(Genre.NO_FILTER);
+        releaseYearPicker.setValue(Year.NO_FILTER);
+        ratingComboBox.setValue(Rating.NO_FILTER);
+
+        List<Movie> allMovies;
+
+        try {
+            allMovies = MovieAPIService.getMovies();
+        } catch (IOException e){
+            allMovies = new ArrayList<>();
         }
-        filteredList.setPredicate(movie -> true);
+
+        observableMovies.addAll(allMovies);
+        sortMovies();
     }
 
-    public static void searchForMovie(String searchTerm, Genre genre, FilteredList<Movie> filteredList, List<Movie> movies) {
-        Set<Movie> searchResults = new HashSet<>();
-        Set<Movie> keywordSearchResults = MovieSearchService.searchKeyword(searchTerm, movies);
-        List<Movie> genreSearchResults = MovieFilterService.filterMoviesByGenre(genre, movies);
+    private void setFilter() {
+        observableMovies.clear();
+
+        List<Movie> moviesWithFilter;
 
-        for (Movie movie : keywordSearchResults) {
-            if (genreSearchResults.contains(movie)) {
-                searchResults.add(movie);
-            }
+        String genre = genreComboBox.getValue() == Genre.NO_FILTER ? "" : genreComboBox.getValue().name();
+        String releaseYear = releaseYearPicker.getValue() == Year.NO_FILTER ? "" : releaseYearPicker.getValue().toString();
+        String ratingFrom = ratingComboBox.getValue() == Rating.NO_FILTER ? "" : String.valueOf(ratingComboBox.getValue().getRatingFrom());
+
+        try {
+            moviesWithFilter = MovieAPIService.getMoviesBy(searchField.getText(), genre, releaseYear, ratingFrom);
+        } catch (IOException e){
+            moviesWithFilter = new ArrayList<>();
         }
 
-        filteredList.setPredicate(searchResults::contains);
+        observableMovies.addAll(moviesWithFilter);
+        sortMovies();
     }
 
-    private List<Integer> getYears() {
-        int currentYear = java.time.LocalDate.now().getYear();
-        return IntStream.rangeClosed(0, 80)
-                .mapToObj(i -> currentYear - i)
-                .collect(Collectors.toList());
+    private void sortMovies(){
+        if (sortBtn.getText().equals(SORT_DEFAULT_TEXT_ASC)) {
+            observableMovies.sort(Collections.reverseOrder());
+        } else {
+            observableMovies.sort(Comparator.naturalOrder());
+        }
+    }
+
+    private void sortMoviesAndSetButtonText(){
+        if (sortBtn.getText().equals(SORT_DEFAULT_TEXT_ASC)) {
+            observableMovies.sort(Comparator.naturalOrder());
+            sortBtn.setText(SORT_DEFAULT_TEXT_DESC);
+        } else {
+            observableMovies.sort(Collections.reverseOrder());
+            sortBtn.setText(SORT_DEFAULT_TEXT_ASC);
+        }
```

Comment:
These two methods are pretty much the same and `sortMoviesAndSetButtonText()` is used only in the `initialze()` method. I would delete it and just use something like:
```
sortBtn.setOnAction(actionEvent -> {
            sortMovies();
            if (sortBtn.getText().equals(SORT_DEFAULT_TEXT_ASC)) {
                sortBtn.setText(SORT_DEFAULT_TEXT_DESC);
            } else {
                sortBtn.setText(SORT_DEFAULT_TEXT_ASC);
            }
        });
```

Reply:
You can disregard the example, I just realized that it would introduce a bug. Still I would avoid duplicating the common part of the code, if possible